src/common/hooks/use-token-metadata.ts

```diff
@@ -0,0 +1,25 @@
+import { getAssetNameParts } from '@common/utils';
+import { useApi } from '@common/api/client';
+import { useQuery } from 'react-query';
+
+export function useTokenMetadata(
+  token: string,
+  tokenType: 'non_fungible_tokens' | 'fungible_tokens'
+) {
+  const { address, contract, asset } = getAssetNameParts(token);
+  const { fungibleTokensApi } = useApi();
+  const T_24H_IN_MS = 1000 * 60 * 60 * 24;
```

Comment:
we should probably define these settings in a common place and use them in other queries, we can tackle this in a separate PR.